src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -912,7 +912,7 @@ var ReactDOMFiberComponent = {
           case 'selected':
             break;
           default:
-            extraAttributeNames.add(attributes[i].name);
```

Comment:
This was intentional so that you would compare and display it in the case that it was received. For things like SVG where case matters.

Reply:
I don't think this is right because on SVG we do want things like `<svg foobar="" />` and `<svg fooBar="" />` to show up as an extra attribute.

It might make sense to deal with this at the same time as the "all attributes should be lower-case" warning though.

Reply:
I am not sure that we can, unless we can control for browsers that report casing from `attribute.name` inconsistently. For example, IE Edge reports SVG `fill` and `stroke in all caps:

**Test**: https://codepen.io/nhunzaker/pen/qXRPvY

**Screenshot of IE Edge 15:**

`fill`, `stroke`, and `stroke-dasharray` report in all caps.

![screen shot 2017-09-11 at 6 47 29 pm](https://user-images.githubusercontent.com/590904/30300252-aea3a570-9721-11e7-954e-3d15b1103516.png)

**Screenshot of Chrome:**

![screen shot 2017-09-11 at 6 48 30 pm](https://user-images.githubusercontent.com/590904/30300276-d4d4ae1a-9721-11e7-9847-de4ad62df22b.png)

Reply:
@nhunzaker Can we special case the known ones since they will also have the warning about invalid case of the prop?

Reply:
I can start to compile a list of special cases. I'm sort of worried that this will be tedious and it'll end up being most of SVG, but I don't know that yet. I know if differences in casing of SVG in IE Edge. I didn't know about the cases alluded to in Chrome. 

Should I just open up the attribute table in every browser using [BrowserStack](browserstack.com)? How many versions back should I go? 

Kind of makes me wonder if it could be automated using their remote API.

--- 

I have a few questions, and I apologize in advance if they sound cynical, but I don't want to assume anything so I can better understand the problem:

1: What is the advantage to case-_sensitive_ hydration? Casing warnings for known attributes will raise server-side during markup generation (is this correct?), and then and on update client-side. 

2: Wouldn't the same components consistently misspell attribute names server and client side?

3: Even if the casing is produced differently server/client-side, browsers don't appear to care about casing. I'm curious if this is generally true, or if my tests are flawed. They certainly aren't exhaustive.

Reply:
My concern is that just ignoring case doesn't take the core of the problem in to consideration because it's not the case that browsers completely ignore case. There are specific rules for it.

Specifically, the rules (other than bugs) is that the HTML parser of a server response makes them lower-case except for a whitelist. https://www.w3.org/TR/html5/syntax.html#adjust-svg-attributes

However, this whitelist doesn't apply to client code. Specifically cases like this is what I'm concerned about:

```js
var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
svg.setAttribute('viewbox',  '0 0 1 1');
console.log(svg.viewBox);

var container = document.createElement('div');
container.innerHTML = '<svg viewbox="0 0 1 1"></svg>';
console.log(container.firstChild.viewBox);
```

Rendering the "viewbox" attribute on the server and the client behaves differently because it *is* case-sensitive.

So simple saying that they're not case-sensitive is not the full story. It might be the case that this is the best compromise anyway but I want to make sure we've considered all the aspects because "DOM is case-insensitive" is an oversimplification so we might be missing something.

Reply:
Ah got it. So there _are_ attributes that care about casing. That's frustrating... 

Thank you for spelling this out for me. Prior to this point, I was not aware of the differences between HTML/SVG parsing and the DOM attribute manipulation. What a mess!